### PR TITLE
ember / glimmer: Preserve self-closing tags

### PR DESCRIPTION
**Pull request by dcyriller**
## Description

As reported in https://github.com/prettier/prettier/issues/13624, self-closing tags are supported by Ember ([doc](https://guides.emberjs.com/v4.7.0/components/#toc_self-closing-tags)).

This PR would preserve them. The PR would close https://github.com/prettier/prettier/issues/13624.

## Checklist

- [x] I’ve added tests to confirm my change works.
- [] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/language-handlebars/utils.js [hunk 1]
```diff
@@ -41,6 +41,7 @@ const voidTags = new Set(htmlVoidElements);
 function isVoid(node) {
   return (
     voidTags.has(node.tag) ||
+    node.selfClosing === true ||
```
Comment (fisker) [inline src/language-handlebars/utils.js:44]: Should we check this in print? So we can keep `<i></i>`?

Comment (dcyriller) [reply src/language-handlebars/utils.js:44]: Do you think `<i/>` should be formatted to `<i></i>`? While `<div />` should remain `<div />`? (inline elements vs others)

(Just pushed more tests and verbose changelog to better describe the changes.)

Comment (fisker) [reply src/language-handlebars/utils.js:44]: No, I mean maybe we should keep it as it is. It's not our job to enforce the self close style.

Comment (fisker) [reply src/language-handlebars/utils.js:44]: `<i></i>` -> `<i></i>` 
`<i/>`  -> `<i/>`

Comment (dcyriller) [reply src/language-handlebars/utils.js:44]: oh I see, thanks for the clarification
I have added more test cases in the PR to show it doesn't enforce the self close style (except for the `<Component />` case, but that was already the case prior to this PR, for a reason I don't remember)

Comment (fisker) [reply src/language-handlebars/utils.js:44]: Ah, sorry, I misunderstood.

**Review by thorn0 — APPROVED** [review]

File: src/language-handlebars/utils.js [hunk 2]
```diff
@@ -41,6 +41,7 @@ const voidTags = new Set(htmlVoidElements);
 function isVoid(node) {
   return (
     voidTags.has(node.tag) ||
```
Comment (fisker) [inline src/language-handlebars/utils.js:43]: Should we remove the html void tag check? 

`` should always self closed.  If anyone wrote `</img>`, we can just ignore.

**Comment by fisker** [comment]
I'm going to merge this, and address https://github.com/prettier/prettier/pull/13691#discussion_r1001731467 in another PR.

**Pull request merged**